### DISCUSSION TITLE
Click through from rack elevation to device is missing the port number in the URL

### DISCUSSION THREAD

**Discussion by dunc623b** [netbox-community/netbox#5780] [Q&A]
### Environment
* Python version: 3.7.3
* NetBox version: 2.10.3

### Steps to Reproduce
1. I run Netbox on port 8000, so URLs are of the form https://netbox-lab:8000/
2. This is fine apart from on the Rack page (e.g. /dcim/racks/22/ ), on the rack elevation diagram, the URL to click on the device and view the device page is missing the port (e.g. is of the form https://netbox-lab/dcim/devices/94/ instead of https://netbox-lab:8000/dcim/devices/94/
3. The URL to click on an empty location and go to Add Device is fine and contains the port no problem, it's just the click to device one that is wrong.

### Expected Behavior
Expected the URL to be of the form https://netbox-lab:8000/dcim/devices/94/

### Observed Behavior
URL is missing the port, e.g. https://netbox-lab/dcim/devices/94/

**Comment by jeremystretch** [comment]
This is a problem with your HTTP server configuration. When running NetBox on a non-standard port, the HTTP frontend needs to pass the port number to the WSGI process.

**Reply by dunc623b** [reply]
Thanks for the response. I thought it might be something to do with the proxy but from what I can see every other URL works just fine it's just the one location in rack elevation behaving differently. I'll see if I can work it out. Cheers.

**Reply by candlerb** [reply]
Your proxy needs to set an the `X-Forwarded-Host:` header with the `host:port` that it is being accessed on.  The supplied sample Apache and Nginx [configs](https://github.com/netbox-community/netbox/tree/develop/contrib) show how to do that.

You'll need `X-Forwarded-Proto:` as well for https.

**Reply by dunc623b** [reply]
Hi candlerb,

Thanks very much for the reply, and sorry I've been so long getting round to looking at it.

I did already set the X-Forwarded-Host header, but for some reason I had been using $server_name and thus missing the port.

I've now switched $server_name for $http_host as per the example configs and we're all good.

Thanks very much for your help.

Cheers,

Dunc
